pkg/kubelet/prober/worker.go

```diff
@@ -182,7 +182,7 @@ func (w *worker) doProbe() (keepGoing bool) {
 	}
 
 	// Worker should terminate if pod is terminated.
-	if status.Phase == v1.PodFailed || status.Phase == v1.PodSucceeded {
+	if status.Phase == v1.PodFailed || status.Phase == v1.PodSucceeded || w.pod.DeletionTimestamp != nil {
```

Comment:
Is the check in line 179 not a sufficient check to determine whether the pod has been deleted?

If we do decide to check the `DeletionTimestamp`, I think we should move it into a separate `if` statement. The other two `status` conditions we are checking seem to be checking if the pod is terminated, not deleted.